WordPress/gutenberg
# Quote v2: update how attributes are registered

Follow-up to #39703 and #39704 

## What?

This PR updates how the new attribute for Quote v2 is registered. As a side-effect, the v2 quote block declares all the block supports the v1 has and the `align` attribute as well.

## Why?

Using a separate `block.json` for v2 is problematic because it's not picked up automatically.

## How?

Instead of having a separate `block.json` for v2, this uses the `blocks.registerBlockType` hook to register the new attribute `attribution` and unregister the old one `citation`.

## Testing Instructions

Check that attributes are registered properly is v2 is active:

- Go to "Gutenberg > Experiments" page and enable the v2 of the quote block.
- Load the editor.
- Verify that the `core/blocks` store contains the proper block attributes under `blockTypes.core/quote`:
  -  it should not have the `value` and `citation` attributes
  - it should have the `attribution` attribute

Check that attributes are registered properly if v1 is active:

- Go to "Gutenberg > Experiments" page and disable the v2 of the quote block.
- Load the editor.
- Verify that the `core/blocks` store contains the proper block attributes under `blockTypes.core/quote`:
  -  it should have the `value` and `citation` attributes
  - it should not have the `attribution` attribute
